share/spack/gitlab/cloud_pipelines/stacks/rocm/spack.yaml

```diff
@@ -0,0 +1,155 @@
+spack:
+  view: false
+
+  concretizer:
+    reuse: false
+    unify: true
+
+  config:
+    install_tree:
+      root: /home/software/spack
+      padded_length: 512
+      projections:
+        all: '{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash}'
+
+  definitions:
+    - default_specs:
+      - hip
+      - hipblas
+      - hipcub
+      - hipfft
+      - hipfort
+      - hipify-clang
+      # - hip-rocclr # Needs fix for install
+      - hipsolver
+      - hipsparse
+      # - hipsycl # Needs fixes for to use llvm-amdgpu clang
+      - hsa-rocr-dev
+      - mesa +llvm
+      - miopen-hip
+      - rocalution
+      - rocblas
+      - rocfft
+      - rocm-clang-ocl
+      - rocm-cmake
+      - rocm-dbgapi
+      - rocm-debug-agent
+      # - rocm-device-libs # built by llvm-amdgpu
+      - rocm-gdb
+      - rocminfo
+      - rocm-opencl
+      - rocm-openmp-extras
+      # - rocm-smi # Deprecated after 4.1.0 in favor of rocm-smi-lib
+      - rocm-smi-lib
+      # - rocm-tensile #
+      # - rocm-validation-suite # Needs hip-rocclr
+      - rocprim
+      - rocprofiler-dev
+      - rocrand amdgpu_target=gfx1030
+      - rocsolver
```

Comment:
Unless you're checking the performance of these binaries, I would recommend `rocsolver ~optimal`. It omits all the loop-unrolled kernels that are templated for different problem sizes, so it builds much faster without changing what the library needs from its dependencies or what it provides to its dependents.